internal/lines.go

```diff
@@ -72,6 +77,15 @@ func (lh *LineHandler) Start() {
 				err := lh.Flush()
 				if err != nil {
 					log.Println(err)
+					if err == errThrottled {
```

Comment:
Hmmm, this would impact the other atoms as well (metrics, spans etc) as we are only looking for a 406 status code. We should really try to isolate this to just the event handling.

I would suggest creating a separate ErrorHandler interface that is an optional setter for LineHandler (only leveraged when creating the eventHandler). And LineHandler delegates to this here and in the report method if it is non-nil.

Reply:
this is just to show a log message